iOS/Controllers/Lesson/LessonsViewController.swift

```diff
@@ -71,7 +71,8 @@ public class LessonsViewController: UITableViewController {
             guard let indexPath = tableView.indexPath(for: selectedCell) else { return }
             guard let destination = segue.destination as? SingleLessonViewController else { return }
             let selectedLesson = fetchedResultsController.object(at: indexPath)
-            destination.lesson = selectedLesson
+            destination.content = Array(selectedLesson.contents)
+            destination.title = selectedLesson.name
```

Comment:
I would prefer if the `SingleLessonViewController` is not configured for the outside. just set the lesson and the rest is handled in `SingleLessonViewController`.

Furthermore, transforming the set `selectedLesson.contents` to an array is probably not deterministic, so the order might be different every time. A better approach would be to fetch the content (with an sort predicate) again from CoreData.